pydantic/_internal/_validators.py

```diff
@@ -231,14 +231,14 @@ def __call__(  # noqa: C901 (ignore complexity)
                     {'multiple_of': self.multiple_of},
                 )
 
-        if self.gt is not None and not value > self.gt:
+        if self.gt is not None and not value > self.gt:  # type: ignore
```

Comment:
maybe worth adding a note about why these type ignores are present (something about decimal IIRC?)

Reply:
This one was quite perplexing: 
> Operator ">" not supported for types "<subclass of int and Decimal> | <subclass of float and Decimal> | <subclass of str and Decimal> | Decimal" and "int | Decimal"

I can add that, but to be honest it doesn't even make much sense to me. This is probably a good candidate for exploring more later. We should come back in the future at some sprint and try to clean up type ignores

Reply:
can we add this as a comment to help future travellers passing this way.

Reply:
done